### PR TITLE
Use MKs in `ouroboros-consensus-cardano-tools`

### PR DESCRIPTION
**Pull request by Jasagredo**
# Description

Use the Ledger Tables in both DbAnalyser and DbSynthesizer.

# Checklist

- Branch
    - [ ] The documentation has been properly updated

### CODE REVIEW

**Comment by Jasagredo** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4352** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4352" target="_blank">Graphite</a> 
    * **PR #4341** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4341" target="_blank">Graphite</a> 
      * **PR #4342** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4342" target="_blank">Graphite</a> 
        * **PR #4343** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4343" target="_blank">Graphite</a> 
    * **PR #4316** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4316" target="_blank">Graphite</a> 
    * **PR #4344** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4344" target="_blank">Graphite</a> 
      * **PR #4366** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4366" target="_blank">Graphite</a> 
        * **PR #4363** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4363" target="_blank">Graphite</a> 
          * **PR #4391** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4391" target="_blank">Graphite</a> 
            * **PR #4417** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4417" target="_blank">Graphite</a> 
              * **PR #4424** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4424" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4424?utm_source=stack-comment).

**Comment by Jasagredo** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4352** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4352" target="_blank">Graphite</a> 
    * **PR #4341** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4341" target="_blank">Graphite</a> 
      * **PR #4342** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4342" target="_blank">Graphite</a> 
        * **PR #4343** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4343" target="_blank">Graphite</a> 
    * **PR #4316** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4316" target="_blank">Graphite</a> 
    * **PR #4344** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4344" target="_blank">Graphite</a> 
      * **PR #4366** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4366" target="_blank">Graphite</a> 
        * **PR #4363** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4363" target="_blank">Graphite</a> 
          * **PR #4391** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4391" target="_blank">Graphite</a> 
            * **PR #4417** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4417" target="_blank">Graphite</a> 
              * **PR #4424** <a href="https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4424" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/input-output-hk/ouroboros-network/4424?utm_source=stack-comment).

**Review by dnadales — APPROVED** [review]

File: ouroboros-consensus-cardano-tools/src/Cardano/Tools/DBAnalyser/Analysis.hs [hunk 1]
```diff
@@ -594,16 +655,31 @@ benchmarkLedgerOps mOutfile AnalysisEnv {db, registry, initLedger, cfg, limit} =
...
+        hydrateTheTickedState ::
+             Ticked1 (LedgerState blk) DiffMK
+          -> IO (Ticked1 (LedgerState blk) ValuesMK)
+        hydrateTheTickedState st = do
+          let dbch = ledgerDbChangelog ldb
+              ks = getBlockKeySets blk
+              aks = rewindTableKeySets dbch ks
+          urs <- readKeySets bstore aks
+          case forwardTableKeySets dbch urs of
+            Left err -> error $ "Rewiing;read;forward failed" <> show err
```
Comment (jorisdral) [inline ouroboros-consensus-cardano-tools/src/Cardano/Tools/DBAnalyser/Analysis.hs:667]:
```suggestion
            Left err -> error $ "Rewind;read;forward failed" <> show err
```

**Review by jorisdral — APPROVED** [review]

File: ouroboros-consensus-cardano-tools/src/Cardano/Tools/DBAnalyser/Analysis.hs [hunk 2]
```diff
@@ -444,23 +477,30 @@ traceLedgerProcessing ::
...
+      :: LedgerDB' blk
       -> blk
-      -> IO (ExtLedgerState blk)
+      -> IO (LedgerDB' blk)
     process oldLedger blk = do
-      let ledgerCfg     = ExtLedgerCfg cfg
-          appliedResult = tickThenApplyLedgerResult ledgerCfg blk oldLedger
-          newLedger     = either (error . show) lrResult $ runExcept $ appliedResult
-          traces        =
+      ldb' <- LedgerDB.push (configLedgerDb cfg) (LedgerDB.ReapplyVal blk) (readKeySets bstore) oldLedger
```
[the remark this answers is not in the document]
Comment (Jasagredo) [reply ouroboros-consensus-cardano-tools/src/Cardano/Tools/DBAnalyser/Analysis.hs:486]: I don't think it matters, and we would have to fight lifting if we wanted to Apply

**Pull request merged**